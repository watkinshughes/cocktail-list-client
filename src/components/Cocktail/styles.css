.Cocktail {
    margin: 20px 0;
    padding: 20px;
    background: #fff;
}

.Cocktail h1 {
    font-size: 1.5em;
    font-weight: bold;
}

.Cocktail h2 {
    font-size: 1em;
    font-weight: bold;
}