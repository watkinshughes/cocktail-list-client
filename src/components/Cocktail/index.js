import React, { Component } from "react";
import { Link } from "react-router-dom";
import styles from "./styles.module.css";

class Cocktail extends Component {
  render() {
    return (
      <article className={styles.cocktail}>
        <Link to={this.props.info.id}>
          <h1>{this.props.info.name}</h1>
        </Link>
      </article>
    );
  }
}

export default Cocktail;
