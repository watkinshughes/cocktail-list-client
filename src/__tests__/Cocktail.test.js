import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter as Router } from "react-router-dom";
import { create } from "react-test-renderer";
import Cocktail from "../components/Cocktail";

const info = {
  id: "12345"
};

test("snapsot", () => {
  const c = create(
    <Router>
      <Cocktail info={info} />
    </Router>
  );
  expect(c.toJSON()).toMatchSnapshot();
});

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <Router>
      <Cocktail info={info} />
    </Router>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});
