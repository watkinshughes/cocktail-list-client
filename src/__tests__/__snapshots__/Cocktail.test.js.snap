// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapsot 1`] = `
<article
  className="cocktail"
>
  <a
    href="/12345"
    onClick={[Function]}
  >
    <h1 />
  </a>
</article>
`;
